sklearn/tree/_splitter.pyx

```diff
@@ -131,13 +135,15 @@ cdef class Splitter:
         X : object
             This contains the inputs. Usually it is a 2d numpy array.
 
-        y : numpy.ndarray, dtype=DOUBLE_t
+        y : array-like, dtype=DOUBLE_t
             This is the vector of targets, or true labels, for the samples
 
-        sample_weight : numpy.ndarray, dtype=DOUBLE_t (optional)
+        sample_weight : array-like, dtype=DOUBLE_t
```

Comment:
This is not an array-like but a `DOUBLE_t*` (i.e. a pointer on a DOUBLE_t type)

I don't think we have a convention for these. I guess a reasonable one would be

`sample_weight : pointer on DOUBLE_t`